editor/editor_help.cpp

```diff
@@ -1419,70 +1454,16 @@ static void _add_text_to_rt(const String &p_bbcode, RichTextLabel *p_rt) {
 	DocData *doc = EditorHelp::get_doc_data();
 	String base_path;
 
-	/*p_rt->push_color(EditorSettings::get_singleton()->get("text_editor/highlighting/text_color"));
-	p_rt->push_font( get_font("normal","Fonts") );
-	p_rt->push_indent(1);*/
-	int pos = 0;
-
 	Ref<Font> doc_font = p_rt->get_font("doc", "EditorFonts");
 	Ref<Font> doc_code_font = p_rt->get_font("doc_source", "EditorFonts");
+	Color font_color_hl = p_rt->get_color("highlight_color", "RichTextLabel");
+	Color link_color = p_rt->get_color("accent_color", "Editor").linear_interpolate(font_color_hl, 0.8);
 
 	String bbcode = p_bbcode.replace("\t", " ").replace("\r", " ").strip_edges();
 
-	//change newlines for double newlines
-	for (int i = 0; i < bbcode.length(); i++) {
-
-		//find valid newlines (double)
-		if (bbcode[i] == '\n') {
-			bool dnl = false;
-			int j = i + 1;
-			for (; j < p_bbcode.length(); j++) {
-				if (bbcode[j] == ' ')
-					continue;
-				if (bbcode[j] == '\n') {
-					dnl = true;
-					break;
-				}
-				break;
-			}
-
-			if (dnl) {
-				bbcode[i] = 0xFFFF;
-				//keep
-				i = j;
-			} else {
-				bbcode = bbcode.insert(i, "\n");
-				i++;
-				//bbcode[i]=' ';
-				//i=j-1;
-			}
-		}
-	}
-
-	//remove double spaces or spaces after newlines
-	for (int i = 0; i < bbcode.length(); i++) {
-
-		if (bbcode[i] == ' ' || bbcode[i] == '\n' || bbcode[i] == 0xFFFF) {
-
-			for (int j = i + 1; j < p_bbcode.length(); j++) {
-				if (bbcode[j] == ' ') {
-					bbcode.remove(j);
-					j--;
-					continue;
-				} else {
-					break;
-				}
-			}
-		}
-	}
-
-	//change newlines to double newlines
-
-	CharType dnls[2] = { 0xFFFF, 0 };
-	bbcode = bbcode.replace(dnls, "\n");
-
```

Comment:
@djrm is this double space section the part that caused the error with duplicating the docs in one line?

sometimes I also had selection issues. I guess they are also gone now?

Reply:
@toger5 there were a lot issues, removing all that in that function only solved the codeblock thing, but the overlapping text was caused by a wrong pop call (was hard to find). also toger there were never duplication, if you check your screenshot you will see that it appears to be so, but it is just the way members descriptions are added, it seems that all the methods you saw used the same Setter and Getter, thats why it looked duplicated.

Reply:
I'm pretty sure there were duplicates too.

But yea I also spend time trying to find stuff in the editor_help class but it's super hard (kudos for you !!)

hera are some more screenshots (sorry they take a lot of space...) that show the dulicated sections.
<img width="797" alt="screen shot 2017-09-14 at 04 39 57" src="https://user-images.githubusercontent.com/16718859/30409689-034b28d2-9907-11e7-8c5a-ec9b54d1a7cd.png">
![Uploading Screen Shot 2017-09-14 at 04.39.49.png…]()